src/tests/ftest/security/cont_get_acl.py

```diff
@@ -0,0 +1,128 @@
+        self.daos_cmd = self.get_daos_command()
+        self.prepare_pool()
+        self.add_container(self.pool)
+
+    @fail_on(CommandFailure)
+    def test_acl_get_valid(self):
+        """
+        JIRA ID: DAOS-3705
+
+        Test Description: Test that container get-acl command performs as
+            expected with invalid inputs.
+
+        :avocado: tags=all,pr,security,container_acl,cont_get_acl_inputs
+        """
+        test_errs = []
+        for verbose in [True, False]:
+            for outfile in self.params.get("valid_out_filename", "/run/*"):
+                path_to_file = os.path.join(
+                    self.tmp, "{}_{}".format(outfile, verbose))
+
+                # Enable raising an exception if the daos command fails
+                self.daos_cmd.exit_status_exception = False
+                self.daos_cmd.container_get_acl(
+                    self.pool.uuid,
+                    self.pool.svc_ranks[0],
+                    self.container.uuid,
+                    verbose=verbose,
+                    outfile=path_to_file)
+
+                # Verify consistency of acl obtained through the file
+                file_acl = read_acl_file(path_to_file)
+                self.acl_file_diff(file_acl)
+
+                # Let's verify that we can't overwrite an already existing file
+                # Disable raising an exception if the daos command fails
+                self.daos_cmd.exit_status_exception = False
+                self.daos_cmd.container_get_acl(
+                    self.pool.uuid,
+                    self.pool.svc_ranks[0],
+                    self.container.uuid,
+                    verbose=verbose,
+                    outfile=path_to_file)
+                test_errs.extend(
+                    self.error_handling(
+                        self.daos_cmd.result, "File already exists"))
+
+        if test_errs:
+            self.fail("container get-acl command expected to fail: \
+                {}".format("\n".join(test_errs)))
+
+    def test_no_user_permissions(self):
+        """
+        JIRA ID: DAOS-3705
+
+        Test Description: Test that container get-acl command doesn't
+            get ACL information without permission.
+
+        :avocado: tags=all,pr,security,container_acl,cont_get_acl_noperms
+        """
+        # Let's give access to the pool to the root user
+        self.get_dmg_command().pool_update_acl(
+            self.pool.uuid, entry="A::EVERYONE@:rw")
+
+        # The root user shouldn't have access to getting container ACL entries
+        self.daos_cmd.sudo = True
+
+        # Disable raising an exception if the daos command fails
+        self.daos_cmd.exit_status_exception = False
+
+        # Let's check that we can't run as root (or other user) and get
+        # acl information if no permissions are set for that user.
+        test_errs = []
+        for verbose in [True, False]:
+            for outfile in self.params.get("valid_out_filename", "/run/*"):
```

Comment:
Probably not necessary to do this loop again in this case.

Reply:
Yeah, it's technically not necessary. The reason I implemented the loop in this case as well was to be thorough in checking that the valid inputs that should pass on the previous case all fail for this case, provided that the user has no permissions. I can definitely change this to check for the permission error in a way that isn't doing the loop again.